### ISSUE TITLE
Customer export don't work well (Export only a few records, 20 to be exact, ID from 2 to 21)

### ISSUE THREAD

**Issue by iw0der** [magento/magento2#3536]
Hi,
I finished loading about 500 customers and then try to export but the exported file stops on the 21° row!

![customersexport](https://cloud.githubusercontent.com/assets/16407046/13296546/56cd50b8-db2e-11e5-83c8-0e704b2878e6.JPG)

The file contains only the header with field names and 20 customers with ID from 2 to 21; while I was expecting to find 445 :-(

I also tried to change the number of records per page, to turn pages and export, but the procedure always and only extracts the first 20 customers and I can not rule it otherwise!

I also tried using _System > Export_ and find such a result, even before confirming the export:
- if I choose "Customers Main File", then the system after an instant writes "22 records found"
- if I choose "Customer Addresses", writes "19 records found" 

But in any case, I can browse and view all clients entered in the test on _Customers > All Customers_

In short, customers are not 19, 20, 22 it even less but 445! How to export them all?

So, this happens with MG CE v. 2.0.2.

Thanks,
A.

**Comment by BaDos** [comment]
Hi @iw0der 

Thank you for reporting.

I cannot reproduce this bug on develop branch.

But there is still the bug of exporting on customers' grid on Magento 2.0.9. Internal ticket MAGETWO-58296 was created.
System > Export/Import customers works correctly.

**Comment by philippsander** [comment]
FYI, this is still an issue in CE 2.0.10

**Comment by magento-engcom-team** [comment]
@iw0der, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
We tested the issue on 2.3.0-dev, 2.2.0, 2.1.9

**Comment by vincentwinkel** [comment]
Same issue in 2.0.14. Please learn how to work and stop closing tickets all the time saying it's fixed in 2.1+, we don't care, we use the 2.0.
